### PR TITLE
Add TypeScript declarations to app-storage.

### PR DESCRIPTION
**Pull request by aomarks**
This PR adds TypeScript declarations generated by https://github.com/Polymer/gen-typescript-declarations/

These declarations can be re-generated by running `npm run update-types`.

Tracker: https://github.com/Polymer/gen-typescript-declarations/issues/79

### CODE REVIEW

File: app-storage-behavior.d.ts [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    /**
+     * When true, will perform detailed logging.
+     */
+    log: boolean|null|undefined;
+
+    /**
+     * Override this getter to return true if the value has never been
+     * persisted to storage.
+     */
+    readonly isNew: any;
```
Comment (e111077) [inline app-storage-behavior.d.ts]: This getter has a `@type` closure annotation.

https://github.com/PolymerElements/app-storage/blob/master/app-storage-behavior.html#L87

File: app-storage-behavior.d.ts [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+     * A promise that will resolve once all queued transactions
+     * have completed.
+     *
+     * This field is updated as new transactions are enqueued, so it will
+     * only wait for transactions which were enqueued when the field
+     * was accessed.
+     *
+     * This promise never rejects.
+     */
+    readonly transactionsComplete: any;
```
Comment (e111077) [inline app-storage-behavior.d.ts]: this getter also has an `@type` closure annotation

Comment (aomarks) [reply app-storage-behavior.d.ts]: sadly the annotation for getters is `@return`, so we'll need to update this source.

Comment (e111077) [reply app-storage-behavior.d.ts]: 👍

Comment (aomarks) [reply app-storage-behavior.d.ts]: Done, updated annotation in source.

File: app-storage-behavior.d.ts (hunk 1 above)
Comment (aomarks) [reply app-storage-behavior.d.ts]: Done, updated annotation in source.

**Pull request merged**